apps/myaccount/src/components/account-recovery/options/sms-recovery/sms-recovery.tsx

```diff
@@ -0,0 +1,435 @@
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t(
+                            "myAccount:components.accountRecovery.SMSRecovery." +
+                            "notifications.updateMobile.error.message"
+                        )
+                    });
+
+                    return;
+                }
+
+                onAlertFired({
+                    description: t(
+                        "myAccount:components.accountRecovery.SMSRecovery." +
+                        "notifications.updateMobile.genericError.description"
+                    ),
+                    level: AlertLevels.ERROR,
+                    message: t(
+                        "myAccount:components.accountRecovery.SMSRecovery." +
+                        "notifications.updateMobile.genericError.message"
+                    )
+                });
+            });
+    };
+
+    /**
+     * This function gets the mobile number from the response passed as the argument
+     * and assigns it to mobile and editedMobile.
+     * @param response - response as the parameter.
+     * Temporarily the first mobile type element in the phoneNumbers array is shown.
+     * In the future, we need to decide whether or not to allow multiple recovery mobiles.
+     */
+    const setMobileNumber = (response: BasicProfileInterface) => {
+        let mobileNumber: string = "";
+
+        if (response.phoneNumbers) {
+            if (typeof response.phoneNumbers[0] === "object") {
+                const mobileValue : MultiValue = response.phoneNumbers.find(
+                    (phoneNumber : MultiValue) => phoneNumber.type === "mobile");
+
+                mobileNumber = mobileValue?.value;
+                mobileType = mobileValue?.type;
+            } else {
+                mobileNumber = response.phoneNumbers[0] as string;
+                mobileType = "array";
+            }
+        }
+        setMobile(mobileNumber);
+        setEditedMobile(mobileNumber);
+    };
+
+    useEffect(() => {
+        if (!isEmpty(profileInfo)) {
+            setMobileNumber(profileInfo);
+        }
+    }, [ profileInfo ]);
+
+    /**
+     * This is called when the edit icon is clicked.
+     *
+     */
+    const handleEdit = () => {
+        dispatch(setActiveForm(CommonConstants.SECURITY + SMS));
+    };
+
+    /**
+     * This is called when the cancel button is pressed
+     */
+    const handleCancel = () => {
+        dispatch(setActiveForm(null));
+    };
+
+    /**
+     * This function masks the mobile number passed as the argument and returns the masked mobile number.
+     * The text between the third character of the mobile and the second last character is masked.
+     * @param mobileNumber - mobile number.
+     */
+    const maskMobile = (mobileNumber: string) => {
+        return mobileNumber.slice(0,3) + "*".repeat(mobileNumber.length - 5) + mobileNumber.slice(-2);
```

Comment:
change into one line and add a return type.

Reply:
check all function definitions

Reply:
Addressed with https://github.com/wso2/identity-apps/pull/6418/commits/0dd77c8c16b11b96a91e6b1331294e3bb8560a5b